pkg/util/kubernetes/autoscalers/datadogexternal.go

```diff
@@ -185,8 +185,15 @@ func (p *Processor) updateRateLimitingMetrics() error {
 
 // NewDatadogClient generates a new client to query metrics from Datadog
 func NewDatadogClient() (*datadog.Client, error) {
-	apiKey := config.SanitizeAPIKey(config.Datadog.GetString("api_key"))
-	appKey := config.Datadog.GetString("app_key")
+	apiKey := config.SanitizeAPIKey(config.Datadog.GetString("external_metrics_provider.api_key"))
+	if apiKey == "" {
+		apiKey = config.SanitizeAPIKey(config.Datadog.GetString("api_key"))
+	}
+
+	appKey := config.Datadog.GetString("external_metrics_provider.app_key")
+	if appKey == "" {
+		appKey = config.Datadog.GetString("app_key")
```

Comment:
might be worth adding sanitization on the app key as well?

Reply:
I thought it was some kind of specific process for APIKey, but checking the underlying function, it should be safe to add to `app_key`, done.